### DISCUSSION TITLE
ESP32 port

### DISCUSSION THREAD

**Discussion by alexpeissel** [erichelgeson/BlueSCSI-v1#9] [Ideas]
As the Arduino core is also available in the ESP32, would it be possible to do a port?  

I believe the ESP32 is faster than the STM32F103C, (though I can’t speak to port manipulation).  It is also dual core and loaded with WiFi and Bluetooth.  One core could handle the SCSI emulation, while the other serves up a web interface, allowing drag/drop file uploads and downloads to the disk.

Ethernet over SCSI possibilities would also be amazing; a wireless networked SE/30 is the stuff of dreams :).

**Comment by erichelgeson** [comment]
Ha! that would be neat. RaSCSI does have daynaport emulation working in the develop branch if you're interested. For now I wont be focusing on this but if someone was able to get it to work I'd happily buy an ESP32 to test and merge it in!

**Comment by WarlockD** [comment]
Might have to add level converters though.  ESP32 is NOT 5v tolerant and I have burned out pins for that proof.   However I have run SD card speed tests and  can easily get 8 megabytes a second off the interface so it might be worth the try.

**Comment by marciot** [comment]
I'm a software developer with quite a bit of experience with the ESP32 (and Macintosh programming back in days) and maybe I will experiment with this. I was very torn about buying a BlueSCSI kit vs. a RaSCSI kit, because both projects seemed amazing! However, in the end, I ordered a RaSCSI kit because of the web support. Interestingly, the RaSCSI has level shifters and a big ol' header for the 3.3v Rasp Pi. Maybe with some strategically placed jumper wires I can wire it up to an ESP32 and try porting the BlueSCSI code to it.

Although I have to ask, is there any risk of blowing out a vintage Macintosh if the software is implemented incorrectly? I would hate to try this and fry my Mac Plus...

**Reply by WarlockD** [reply]
Not really.  Honestly the NCR 5380 is fairly robust and can sink quite a bit of power.  I have both built the BlueSCSI and have one of the older  SCSI2SD adaptors.  I would go toward the the RaSCSI as well just for the better interface as well.

**Reply by madrulane** [reply]
@marciot did you try? Thanks.

**Reply by erichelgeson** [reply]
Original question was from 4 years ago, but now BlueSCSI v2 has DaynaPORT emulation and a Pico1/2 is pretty reliability sourced and inexpensive. Not to dissuade anyone from working on a ESP32 port if they want, but in 2025 we have some good options.

**Reply by marciot** [reply]
> @marciot did you try? Thanks.

@mardulas No, I never did try it.
